File: couchbase-persistence/couchbase-persistence-java-mvn/hello-api/src/main/java/com/lightbend/lagom/samples/hello/api/HelloService.java
package com.lightbend.lagom.samples.hello.api;

import akka.Done;
import akka.NotUsed;
import com.lightbend.lagom.javadsl.api.Descriptor;
import com.lightbend.lagom.javadsl.api.Service;
import com.lightbend.lagom.javadsl.api.ServiceCall;
import com.lightbend.lagom.javadsl.api.deser.PathParamSerializers;

import java.util.List;

import static com.lightbend.lagom.javadsl.api.Service.named;
import static com.lightbend.lagom.javadsl.api.Service.pathCall;

public interface HelloService extends Service {

    ServiceCall<NotUsed, String> hello(String id);

    ServiceCall<GreetingMessage, Done> useGreeting(String id);

    ServiceCall<NotUsed, List<UserGreeting>> userGreetings();

    @Override
    default Descriptor descriptor() {
        return named("hello").withCalls(
            pathCall("/api/hello/:id", this::hello),
            pathCall("/api/hello/:id", this::useGreeting),
            pathCall("/api/user-greetings", this::userGreetings)
        )
            .withAutoAcl(true);
    }
}

File: couchbase-persistence/couchbase-persistence-scala-sbt/hello-api/src/main/scala/com/lightbend/lagom/samples/hello/api/HelloService.scala
package com.lightbend.lagom.samples.hello.api

import akka.{Done, NotUsed}
import com.lightbend.lagom.scaladsl.api.{Service, ServiceCall}
import play.api.libs.json.{Format, Json}

import scala.collection.immutable.Seq

trait HelloService extends Service {

  def hello(id: String): ServiceCall[NotUsed, String]

  def useGreeting(id: String): ServiceCall[GreetingMessage, Done]

  def userGreetings(): ServiceCall[NotUsed, Seq[UserGreeting]]

  override final def descriptor = {
    import com.lightbend.lagom.scaladsl.api.Service._
    named("hello")
      .withCalls(
        pathCall("/api/hello/:id", hello _),
        pathCall("/api/hello/:id", useGreeting _),
        pathCall("/api/user-greetings", userGreetings _)
      )
      .withAutoAcl(true)
  }
}

case class GreetingMessage(message: String)

object GreetingMessage {
  implicit val format: Format[GreetingMessage] = Json.format[GreetingMessage]
}

case class UserGreeting(name: String, message: String)

object UserGreeting {
  implicit val format: Format[UserGreeting] = Json.format[UserGreeting]
}
